JotunnLib/Managers/DungeonManager.cs

```diff
@@ -0,0 +1,372 @@
+                        if (customRoom.FixReference)
+                        {
+                            customRoom.Prefab.FixReferences(true);
+                            customRoom.FixReference = false;
+                        }
+
+                        if (customRoom.Theme != 0)
+                        {
+                            RegisterRoomInDungeonDB(customRoom);
+                        }
+                    }
+                    catch (MockResolveException ex)
+                    {
+                        Logger.LogWarning(string.Format("Skipping Room {0}: could not resolve mock {1} {2}", customRoom, ex.MockType.Name, ex.FailedMockName));
+                        toDelete.Add(customRoom.Name);
+                    }
+                    catch (Exception ex2)
+                    {
+                        Logger.LogWarning(string.Format("Exception caught while adding Room: {0}", ex2));
+                        toDelete.Add(customRoom.Name);
+                    }
+                }
+                foreach (string name in toDelete)
+                {
+                    Rooms.Remove(name);
+                }
+
+                DungeonDB.instance.GenerateHashList();
+                GenerateHashes();
+            }
+        }
+
+        private void OnDungeonGeneratorSetupAvailableRooms(DungeonGenerator self)
+        {
+            DungeonGeneratorTheme proxy = self?.gameObject.GetComponent<DungeonGeneratorTheme>();
+
+            if (DungeonGenerator.m_availableRooms != null)
+            {
+                if (proxy != null)
+                {
+                    Logger.LogDebug($"This dungeon generator has a custom theme = {proxy.m_themeName}, adding available rooms");
+
+                    DungeonGenerator.m_availableRooms.AddRange(Rooms.Values
+                        .Where(r => r.Room.m_enabled)
+                        .Where(r => r.ThemeName == proxy.m_themeName)
+                        .Select(r => r.RoomData));
+                }
+                else
+                {
+                    Logger.LogDebug($"Adding additional rooms of type {self.m_themes} to available rooms");
+
+                    DungeonGenerator.m_availableRooms.AddRange(Rooms.Values
+                        .Where(r => r.Room.m_enabled)
+                        .Where(r => self.m_themes.HasFlag(r.Theme))
+                        .Select(r => r.RoomData));
+                }
+
+                if (DungeonGenerator.m_availableRooms.Count <= 0)
+                {
+                    Logger.LogDebug($"DungeonManager's SetupAvailableRooms yielded zero rooms.");
+                }
+            }
+        }
+
+
+        private void InvokeOnVanillaRoomsAvailable()
+        {
+            OnVanillaRoomsAvailable?.SafeInvoke();
+            OnRoomsRegistered?.SafeInvoke();
```

Comment:
`OnRoomsRegistered` has to be invoked after all rooms have been added, i.e. as the last statment of `OnDungeonDBStarted`. Otherwise the two event's would be called at the same time

Reply:
Addressed in https://github.com/Valheim-Modding/Jotunn/pull/434/commits/7c88287485744eb216693d087402bc1139bde7d6 and https://github.com/Valheim-Modding/Jotunn/pull/434/commits/0bc84b4710439a19153ae4cfbc3ceaed9f577f86